### PR TITLE
db: introduce Download API

### PR DESCRIPTION
**Pull request by RaduBerinde**
This commit introduces a `DB.Download()` API which is used during online restore to copy off external data.

### CODE REVIEW

**Review by itsbilal — APPROVED** [review]
LGTM, though I'll let DR chime in if they want this to be more explicitly tied to notions of progress.

**Comment by RaduBerinde** [comment]
Thanks!

**Pull request merged**
